### ISSUE TITLE
Allow implicits for mixfix operators

### ISSUE THREAD

**Issue by vlopezj** [agda/agda#2264]
Here is a proposal:

https://gist.github.com/vlopezj/1de925406d10eba30f6fd849ed174917

TL;DR: Implicits would appear to the right of the part of the mixfix operator they fall under. In particular, this means that sometimes the arguments to a mixfix operator will appear before the implicit argument they depend on.

UPDATE: (per @UlfNorell) Implementing this proposal is complicated by the fact that there are more places where an implicit argument can appear in a type signature than positions in an infix application where an implicit argument can appear (i.e. just after a token). Which position the implicit corresponds to can only be determined once the type of the infix operator is known.

**Comment by guillaumebrunerie** [comment]
I am often annoyed by the impossibility of using implicit arguments with mixfix operators. For instance if I want to add an implicit argument to an existing expression or pattern, I have to convert it to prefix notation, which is annoying and much less readable.

I’m not entirely sure I understand the reason given in the update, is the problem with unnamed implicit arguments? Allowing only named implicit arguments for mixfix operators would already be much better, if it’s easier to implement.

In any case, it would probably make more sense to keep this issue open in the icebox rather that having it as a "wontfix".

**Comment by gallais** [comment]
If you are often inserting the same arguments, you may want to declare a pattern
synonym for the more explicit constructor.

**Comment by andreasabel** [comment]
Agda dev meeting 2024-12-11: new proposal needed.

At least this should work:
```
if_then_else_ : {A : Set} -> Bool -> A -> A -> A

if {A = Nat}  x then y else z

_+_ : ...
 
x + {M = ...} y
```
